web_external/vue/components/DatasetReview/DatasetReviewStore.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * Dataset Review Vuex store.
+ */
+
+import _ from 'underscore';
+import Vue from 'vue';
+
+import { getApiRoot } from 'girder/rest';
+
+import DatasetService from '../../api/dataset';
+
+export const SubmissionState = {
+    UNSUBMITTED: 'unsubmitted',
+    SUBMITTING: 'submitting',
+    SUBMITTED: 'submitted',
+    FAILED: 'failed'
+};
+
+export default {
+    namespaced: true,
+    state: {
+        dataset: null,
+        images: [],
+        submissionState: SubmissionState.UNSUBMITTED,
+        thumbnailWidth: 768
+    },
+    getters: {
+        acceptedImages(state) {
+            return state.images.filter((image) => {
+                return !image.flagged;
+            });
+        },
+        flaggedImages(state) {
+            return state.images.filter((image) => {
+                return image.flagged;
+            });
+        }
+    },
+    mutations: {
+        setDataset(state, dataset) {
+            state.dataset = dataset;
+        },
+        setImages(state, images) {
+            state.images = images;
+        },
+        toggleFlagged(state, index) {
+            Vue.set(state.images[index], 'flagged', !state.images[index].flagged);
+        },
+        setSubmissionState(state, data) {
+            state.submissionState = data;
+        }
+    },
+    actions: {
+        getDataset({ commit }, { id }) {
```

Comment:
That makes sense, but keeping those actions separate would allow for a workflow where the next set of images are loaded after the user submits. The annotation tool already does something similar.